internal/engine/upper_test.go

```diff
@@ -132,17 +145,17 @@ func TestUpper_UpWatchFileChangeThenError(t *testing.T) {
 		f.timerMaker.maxTimerLock.Lock()
 		call := <-f.b.calls
 		assert.Equal(t, service, call.service)
-		assert.Equal(t, []string(nil), call.files)
+		assert.Equal(t, []string{}, call.state.FilesChanged())
```

Comment:
...this eems weird. why not assert len == 0?

Reply:
we want assert.Equal to print the full list of files changed on failure

Reply:
aaaah sure.